cli/tauri-bundler/src/bundle/osx_bundle.rs

```diff
@@ -76,9 +80,207 @@ pub fn bundle_project(settings: &Settings) -> crate::Result<Vec<PathBuf>> {
+    .args(auth_args.clone())
+    .stderr(Stdio::inherit())
+    .output()?;
+
+  if !output.status.success() {
+    return Err(crate::Error::from(format!(
+      "failed to upload app to Apple's notarization servers. {}",
+      std::str::from_utf8(&output.stdout)?
+    )));
+  }
+
+  let regex = Regex::new(r"\nRequestUUID = (.+?)\n")?;
+  let stdout = std::str::from_utf8(&output.stdout)?;
+  if let Some(uuid) = regex.captures_iter(stdout).next() {
+    common::print_info("notarization started; waiting for Apple response...")?;
+    let uuid = uuid[1].to_string();
+    get_notarization_status(uuid, auth_args)?;
+    staple_app(app_bundle_path.clone())?;
+  } else {
+    return Err(crate::Error::from(format!(
+      "failed to parse RequestUUID from upload output. {}",
+      stdout
+    )));
+  }
+
+  Ok(())
+}
+
+fn staple_app(mut app_bundle_path: PathBuf) -> crate::Result<()> {
+  let app_bundle_path_clone = app_bundle_path.clone();
+  let filename = app_bundle_path_clone
+    .file_name()
+    .expect("failed to get bundle filename")
+    .to_str()
+    .expect("failed to convert bundle filename to string");
+
+  app_bundle_path.pop();
+
+  let output = Command::new("xcrun")
+    .args(vec!["stapler", "staple", "-v", filename])
+    .current_dir(app_bundle_path)
+    .stderr(Stdio::inherit())
+    .output()?;
+
+  if !output.status.success() {
+    Err(crate::Error::from(format!(
+      "failed to staple app. {}",
+      std::str::from_utf8(&output.stdout)?
+    )))
+  } else {
+    Ok(())
+  }
+}
+
+fn get_notarization_status(uuid: String, auth_args: Vec<String>) -> crate::Result<()> {
+  std::thread::sleep(std::time::Duration::from_secs(10));
+  let output = Command::new("xcrun")
+    .args(vec!["altool", "--notarization-info", &uuid])
+    .args(auth_args.clone())
+    .stderr(Stdio::inherit())
+    .output()?;
+
+  if !output.status.success() {
+    get_notarization_status(uuid, auth_args.clone())
+  } else {
+    let regex = Regex::new(r"\n *Status: (.+?)\n")?;
+    let stdout = std::str::from_utf8(&output.stdout)?;
+    if let Some(status) = regex.captures_iter(stdout).next() {
+      let status = status[1].to_string();
+      if status == "in progress" {
+        get_notarization_status(uuid, auth_args.clone())
+      } else {
+        if status == "invalid" {
+          Err(crate::Error::from(format!(
+            "Apple failed to notarize your app. {}",
+            std::str::from_utf8(&output.stdout)?
+          )))
+        } else if status != "success" {
+          Err(crate::Error::from(format!(
+            "Unknown notarize status {}. {}",
+            status,
+            std::str::from_utf8(&output.stdout)?
+          )))
+        } else {
+          Ok(())
+        }
+      }
+    } else {
+      get_notarization_status(uuid, auth_args.clone())
+    }
+  }
+}
+
+fn notarize_auth_args() -> crate::Result<Vec<String>> {
+  match (
+    std::env::var_os("APPLE_ID"),
+    std::env::var_os("APPLE_PASSWORD"),
```

Comment:
This is pretty rare, but if you're a member of multiple teams in App Store Connect you need to specify which team you're uploading for. You can pass `--asc-provider` (see https://developer.apple.com/documentation/xcode/notarizing_macos_software_before_distribution/customizing_the_notarization_workflow)